File: frontend/src/types/room.ts
import {User} from "./user";

export enum GameMode {
	Regular = "regular",
}

export interface Room {
	roomId: string,

	name: string,
	players: User[],
	gameSettings: {
		gameMode: GameMode,
		maxPlayers: number,
		isPublic: boolean,
		allowedPlayers?: string[],
	},
}

File: backend/tanktics/src/models/room.py
from enum import Enum
from typing import List

from pydantic import Field
from tanktics.src.models.base_schema import BaseSchema


class RoomGameMode(str, Enum):
    REGULAR = "regular"


class RoomGameSettings(BaseSchema):
    game_mode: RoomGameMode
    max_players: int
    is_public: bool
    allowed_players: List[str] = Field(default_factory=list)


class Room(BaseSchema):
    room_id: str
    name: str
    game_settings: RoomGameSettings
